Mathlib/Order/PrimeSeparator.lean

```diff
@@ -0,0 +1,156 @@
+    let ⟨J, hJ⟩ := hcNe
+    refine ⟨Order.isIdeal_sUnion_of_isChain (fun _ hJ ↦ (hcS hJ).1) hcC hcNe,
+            ⟨le_trans (hcS hJ).2.1 (le_sSup hJ), fun J hJ ↦ (hcS hJ).2.2⟩⟩
+
+  -- Thus, by Zorn's lemma, we can pick a maximal ideal J in S.
+  have zorn := zorn_subset_nonempty S chainub I IinS
+  have hJ := Exists.choose_spec zorn
+  set Jset := Exists.choose zorn
+  let ⟨⟨Jidl, IJ, JF⟩, ⟨_, Jmax⟩⟩ := hJ
+  set J := IsIdeal.toIdeal Jidl
+  use J
+  have IJ' : I ≤ J := IJ
+
+  clear chainub IinS
+
+  -- By construction, J contains I and is disjoint from F. It remains to prove that J is prime.
+  refine ⟨?_, ⟨IJ, JF⟩⟩
+
+  -- First note that J is proper: ⊤ ∈ F so ⊤ ∉ J because F and J are disjoint.
+  have Jpr : IsProper J := isProper_of_not_mem (Set.disjoint_left.1 JF F.top_mem)
+
+  -- Suppose that a₁ ∉ J, a₂ ∉ J. We need to prove that a₁ ⊔ a₂ ∉ J.
+  rw [isPrime_iff_mem_or_mem]
+  intros a₁ a₂
+  contrapose!
+  intro ⟨ha₁, ha₂⟩
+
+  -- Consider the ideals J₁, J₂ generated by J ∪ {a₁} and J ∪ {a₂}, respectively.
+  let J₁ := J ⊔ principal a₁
+  let J₂ := J ⊔ principal a₂
+
+  -- For each i, Jᵢ is an ideal that contains aᵢ, and is not equal to J.
+
+  have a₁J₁ : a₁ ∈ J₁ := mem_of_subset_of_mem (le_sup_right : _ ≤ J ⊔ _) mem_principal_self
+  have a₂J₂ : a₂ ∈ J₂ := mem_of_subset_of_mem (le_sup_right : _ ≤ J ⊔ _) mem_principal_self
+  have J₁J : ↑J₁ ≠ Jset := by refine ne_of_mem_of_not_mem' a₁J₁ ha₁
+  have J₂J : ↑J₂ ≠ Jset := by refine ne_of_mem_of_not_mem' a₂J₂ ha₂
+
+  -- Therefore, since J is maximal, we must have Jᵢ ∉ S.
+  have J₁S : ↑J₁ ∉ S := fun h => J₁J (Jmax J₁ h (le_sup_left : J ≤ J₁))
+  have J₂S : ↑J₂ ∉ S := fun h => J₂J (Jmax J₂ h (le_sup_left : J ≤ J₂))
+
+  -- Since Jᵢ is an ideal that contains I, we have that Jᵢ is not disjoint from F.
+  have J₁F : ¬ (Disjoint (F : Set α) J₁) := by
+    intro hdis
+    apply J₁S
+    simp only [le_eq_subset, mem_setOf_eq, SetLike.coe_subset_coe, S]
+    exact ⟨J₁.isIdeal, le_trans IJ' le_sup_left, hdis⟩
+
+  have J₂F : ¬ (Disjoint (F : Set α) J₂) := by
+    intro hdis
+    apply J₂S
+    simp only [le_eq_subset, mem_setOf_eq, SetLike.coe_subset_coe, S]
+    exact ⟨J₂.isIdeal, le_trans IJ' le_sup_left, hdis⟩
+
+  -- Thus, pick cᵢ ∈ F ∩ Jᵢ.
+  let ⟨c₁, ⟨c₁F, c₁J₁⟩⟩ := Set.not_disjoint_iff.1 J₁F
+  let ⟨c₂, ⟨c₂F, c₂J₂⟩⟩ := Set.not_disjoint_iff.1 J₂F
+
+  -- Using the definition of Jᵢ, we can pick bᵢ ∈ J such that cᵢ ≤ bᵢ ⊔ aᵢ.
+  let ⟨b₁, ⟨b₁J, cba₁⟩⟩ := (mem_ideal_sup_principal a₁ c₁ J).1 c₁J₁
+  let ⟨b₂, ⟨b₂J, cba₂⟩⟩ := (mem_ideal_sup_principal a₂ c₂ J).1 c₂J₂
+
+  -- Since J is an ideal, we have b := b₁ ⊔ b₂ ∈ J.
+  let b := b₁ ⊔ b₂
+  have bJ : b ∈ J := sup_mem b₁J b₂J
+
+  have ineq : c₁ ⊓ c₂ ≤ b ⊔ (a₁ ⊓ a₂) :=
+  -- Now we compute: b ⊔ (a₁ ⊓ a₂) = (b ⊔ a₁) ⊓ (b ⊔ a₂) by distributivity
+  -- and b ⊔ aᵢ ≥ bᵢ ⊔ aᵢ by definition of b and monotonicity of ⊔.
+  -- Thus, b ⊔ (a₁ ⊓ a₂) ≥ (b₁ ⊔ a₁) ⊓ (b₂ ⊔ a₂) ≥ c₁ ⊓ c₂,
+  -- with the last inequality holding by monotonicity of inf.
```

Comment:
You have commented this proof quite nicely but at this point it's slightly too literal with the actual code. I wouldn't include these comments, or perhaps shorten them to a single sentence explaining why this `ineq` will be useful.
```suggestion
```

Reply:
Thanks, I agree, I now reformulated it to emphasize the fact that this is where distributivity is used (as, mathematically speaking, this is the important point in the proof)